migrations/versions/1090_fix_external_ids.py

```diff
@@ -0,0 +1,47 @@
+"""Fix external IDs that were populated into audit events incorrectly
+
+Revision ID: 1090
+Revises: 1080
+Create Date: 2017-12-12 09:10:00.00000
+
+"""
+from alembic import op
+import json
+import sqlalchemy as sa
+from sqlalchemy.sql import text
+
+# revision identifiers, used by Alembic.
+revision = '1090'
+down_revision = '1080'
+
+
+def upgrade():
+    # Grab existing external IDs and populate the translation table
+    project_id_internal_to_external = {}
+    project_prefill_connection = op.get_bind()
+    for internal_id, external_id in project_prefill_connection.execute('SELECT id, external_id FROM direct_award_projects'):
+        project_id_internal_to_external[internal_id] = external_id
+
+    # For audit events that have stored an internal project ID (represented by IDs less than 1 million), update the data
+    # using the translation table above to convert it to its external ID.
+    audit_connection = op.get_bind()
+    for audit_id, audit_data, project_id in audit_connection.execute(
+            "SELECT id, data, object_id "
+            "FROM audit_events "
+            "WHERE type='create_project' AND (data->>'projectExternalId')::bigint < 1000000"
+    ):
+        if project_id:
```

Comment:
Would we ever expect there to be no project_id?

Reply:
No.